AUTHENTICATION.md

```diff
@@ -0,0 +1,330 @@
+    - HTTP request contains:
+        * _accept-language: zh-TW, zh-CN;q=0.5_
+        * _Content-Type: text/xml_
+        * _ACCEPT-LANGUAGE: en;q=0.1_
+    - Message to Sign contains:
+        * _Accept-Language: zh-TW, zh-CN;q=0.5_
+        * _Accept-Language: en;q=0.1_
+        * _Content-Type: text/xml_
+
+For more examples of signing headers, see the [Test Vectors](#test-vectors) appendix.
+
+## Calculating the signature
+Once you have the Message to Sign and the rest of the Authorization header parameters filled out, calculating the signature is a simple matter of:
+```
+secretKey = lookupKey(Authorization[key-id])
+Authorization[signature] = hex(hmac_sha256(secretKey, messageToSign))
+```
+
+This signature should be places in the _signature_ parameter of the Authorization header to allow verification by the remote peer.
+
+# Signature Verification
+
+All messages received from the remote peer should be verified as correctly signed before they are processed, unless the API in question calls for otherwise.
+
+The verification steps are as follows:
+* Check if an **Authorization** (request) or **X-SignedResponse** header is present.
+    - If the header is absent, the message should be rejected
+* Parse the authorization header. The message should be rejected if:
+    - It is malformed
+    - Required fields as given in the [Header Format](#header-format) table above are missing, or their values are malformed.
+* Check if the **timestamp** given in the Authorization header is valid
+    - Reject if the timestamp is older than the configured limit (usually 5 minutes in the past)
+    - Reject if the timestamp is newer than the configured limit (usually 5 minutes into the future)
+    - I.e. Reject if _abs(currentTimeSeconds() - Authorization[timestamp]) > 5 * 60_
+* Locate the key corresponding to the given **partner-id** and **key-id** parameters
+    - Reject if the specified key is unknown.
+* Calculate the signature of the received message using the same steps as detailed in [Signature Generation Process](#signature-generation-process).
+    - Reject if the calculated signature does not equal the signature supplied in the authorization header.
+
+If the signature matches, the request or response may be processed. In the case of a request, care should be taken to check that the partner-id referenced in the header really is authorized to perform the requested operation.
+
+# Test Vectors
+
+This section contains various examples of correctly signed HTTP requests and responses, designed to allow verification of an implementation of this specification. The secret key value used for these examples is the ASCII value "secret_key_change_me".
+
+Only those types of requests which are used by the higher level application level API are required to be implemented for things to work, but to be fully compliant with this specification all of the below must produce matching signatures.
```

Comment:
Not sure if this is a typo, is 'higher level application level API' supposed to be correct, or should it be 'application level API' only?